packages/@aws-cdk/aws-lambda/test/lambda-insights.test.ts

```diff
@@ -0,0 +1,195 @@
+    });
+
+    // AF-SOUTH-1 exists, 1.0.54.0 exists, but 1.0.54.0 isn't supported in AF-SOUTH-1
+    new lambda.Function(stack, 'BadVersion', {
+      code: new lambda.InlineCode('foo'),
+      handler: 'index.handler',
+      runtime: lambda.Runtime.NODEJS_10_X,
+      insightsVersion: lambda.LambdaInsightsVersion.VERSION_1_0_54_0,
+    });
+
+    // On synthesis it should throw an error
+    expect(() => app.synth()).toThrow();
+  });
+
+  test('using a specific version without providing a region', () => {
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, 'Stack', {});
+
+    new lambda.Function(stack, 'MyLambda', {
+      code: new lambda.InlineCode('foo'),
+      handler: 'index.handler',
+      runtime: lambda.Runtime.NODEJS_10_X,
+      insightsVersion: lambda.LambdaInsightsVersion.VERSION_1_0_54_0,
+    });
+
+    // Should be looking up a mapping
+    expect(stack).toHaveResource('AWS::Lambda::Function', {
+      Properties:
+      {
+        Code: { ZipFile: 'foo' },
+        Handler: 'index.handler',
+        Role: { 'Fn::GetAtt': ['MyLambdaServiceRole4539ECB6', 'Arn'] },
+        Runtime: 'nodejs10.x',
+        Layers: [{
+          'Fn::FindInMap': [
+            'CloudWatchLambdaInsightsVersions10540',
+            {
+              Ref: 'AWS::Region',
+            },
+            'arn',
+          ],
+        }],
+      },
+      DependsOn: ['MyLambdaServiceRole4539ECB6'],
+    }, ResourcePart.CompleteDefinition);
+
+    // On synthesis it should not throw an error
+    expect(() => app.synth()).not.toThrow();
+  });
+
+  // Here we're error checking the code which verifies if the mapping exists already
+  test('can create two functions in a region agnostic stack with the same version', () => {
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, 'Stack', {});
+
+    new lambda.Function(stack, 'MyLambda1', {
+      code: new lambda.InlineCode('foo'),
+      handler: 'index.handler',
+      runtime: lambda.Runtime.NODEJS_10_X,
+      insightsVersion: lambda.LambdaInsightsVersion.VERSION_1_0_54_0,
+    });
+
+    new lambda.Function(stack, 'MyLambda2', {
+      code: new lambda.InlineCode('foo'),
+      handler: 'index.handler',
+      runtime: lambda.Runtime.NODEJS_10_X,
+      insightsVersion: lambda.LambdaInsightsVersion.VERSION_1_0_54_0,
+    });
+
+    expect(stack).toHaveResource('AWS::Lambda::Function', {
+      Properties:
+      {
+        Code: { ZipFile: 'foo' },
+        Handler: 'index.handler',
+        Role: { 'Fn::GetAtt': ['MyLambda1ServiceRole69A7E1EA', 'Arn'] },
+        Runtime: 'nodejs10.x',
+        Layers: [{
+          'Fn::FindInMap': [
+            'CloudWatchLambdaInsightsVersions10540',
+            {
+              Ref: 'AWS::Region',
+            },
+            'arn',
+          ],
+        }],
+      },
+      DependsOn: ['MyLambda1ServiceRole69A7E1EA'],
+    }, ResourcePart.CompleteDefinition);
+
+    expect(stack).toHaveResource('AWS::Lambda::Function', {
```

Comment:
I would suggest replacing this with a `toMatchTemplate()` method to checks both the lambda function and that the mapping is part of the template.

Reply:
Wouldn't that be covered by the integ test? Or is there a reason to have it in here as well?

Reply:
I usually prefer to cover this in unit test, but it's fine to leave it as is if you feel that's cleaner.

Reply:
Even more ideal would be to have a method like so -

`expect(stack).toHaveMapping(...)`

Not expecting this to be added here.

Reply:
That would be new functionality in @aws-cdk/asset-internal/jest ?

Reply:
yeah it would be. But let's make that a separate pull request if you're looking to add.

Reply:
Done